### PR TITLE
fix(sekoiaio/cti): return None for unknown indicator

### PR DESCRIPTION
**Pull request by PierrickV**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

Related to PR https://github.com/demisto/content/pull/30345 reviewed by @sapirshuker, @Ni-Knight, @MLainer1, @ilaner, @dorschw 

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by sapirshuker** [comment]
Hi @PierrickV , Thank you for your contribution. Your PR review will be published soon,
In the meantime, Please merge from master and bump the pack version in the pack_metadata.json file and add notes describing your changes to a new Release Notes file. To do that, please run the demisto-sdk update-release-notes command - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#update-release-notes.

**Comment by PierrickV** [comment]
This is the demo video: https://drive.google.com/file/d/1dlBk1wsSd6VGX8dv2AaI1Qpp6nUH6vjt/view?usp=sharing

During the recording I found that:
- !GetIndicatorContext value="aaaaaaa" type="file" : `aaaaaaa` is not a valid hash so it is not stored as indicator
- !file file="aaaaaa": `aaaaaaa` is not a valid hash so it is not stored as indicator
- !file file="e1d4d2e829885b322f7e619cbfc2615f": this hash is unknown on our side, I will make a small record with a working one

**Comment by sapirshuker** [comment]
Hey @PierrickV 
Great job! We need to add one more minor thing according to the standard:
https://xsoar.pan.dev/docs/integrations/dbot#score-types
we need to add a message "No results found"

**Comment by PierrickV** [comment]
Hello @sapirshuker,

In the latest commit, I added your review suggestion. Thank you for taking the time to review my work ! :smile: 

I also tried to update the Docker image & the release note to make the CI happy =)

Have a nice day

**Comment by sapirshuker** [comment]
@PierrickV Hey, Thank you very much for your fix, Great Job! 
We want to be align with all the entries, if we don't have any outputs from the API we don't want to return the indicator in the entry/outputs. In addition, please see my slack messages. Thanks! 
Thanks again for contributing to our repo
Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

**Comment by sapirshuker** [comment]
Hey @PierrickV , According to the demo, two commandresult entries were returned when executing reputation commands. 
For the first commandresult entry, we need no result message when there is no response for the indicator from the API. As the standard here: https://xsoar.pan.dev/docs/integrations/dbot#score-types
In the case of an unknown indicator, verify that the same message appears in both entries.
The function create_indicator_result_with_dbotscore_unknown from the common server python includes an example of aligning with the standard and creating the correct commandresult entry.  
Please feel free to reach out to me with any questions - I'm available here or on slack :)

**Comment by PierrickV** [comment]
@sapirshuker thank you for your kind review :+1: 

I did not fully get what you want me to modify here, Should I:
- keep the first output but add the sentence `"No results found"` to the it ?
- remove it because it was not in the previous version

Have a nice day :sun_with_face:

**Comment by sapirshuker** [comment]
Hey @PierrickV You need to keep the entry and add the message to the entry you can see an example for creation the entry by the standard in the function create_indicator_result_with_dbotscore_unknown from the common server python. 
 Please feel free to reach out to me with any questions - I'm available here or on slack :)

**Comment by sapirshuker** [comment]
Hey @PierrickV Great Job!! Thank you very match for you contribution and for your patience.
I suggest some changes according to the standard, please see my message in Slack. 
Thanks.

**Comment by sapirshuker** [comment]
Hi @PierrickV , we haven’t heard from you in a while.
Do you need any help with the pull request?

Please feel free to reach out to me here or on Slack.
Thanks again for contributing to our repo, hope to hear from you soon.

**Comment by PierrickV** [comment]
Hello @sapirshuker,

I reviewed :eyes:  and :heavy_check_mark: added your changes to this PR

**Comment by sapirshuker** [comment]
Hey @PierrickV 
Great job! just a minor change in order save BC: 
change the line: `object_reputation.readable_output = "No results found."`
```
object_reputation.readable_output = tableToMarkdown(name=f'{INTEGRATION_NAME}:',
                                      t={indicator["type"]: indicator["value"], 'Result': 'Not found'},
                                      headers=[indicator_type, 'Result'])
```
 you can also change the 671 line back to `markdown = f"### {indicator['value']} of type {indicator['type']} is an unknown indicator.`
In addition, please fix the CI errors. 
Please feel free to reach out to me here or on Slack.

**Review by sapirshuker — COMMENTED** [review]
Great job!!

**Comment by PierrickV** [comment]
Hello @sapirshuker, :smile: 

I added a new commit to bump the version and fix the tests following the latest review. 

This PR should now pass the CI

File: Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py [hunk 1]
```diff
@@ -252,34 +261,38 @@ def get_ip_indicator_reputation(stix_object: dict, reputation_score: int, reliab
...
 
-def get_file_indicator_reputation(stix_object: dict, reputation_score: int, reliability_score: str, tlp: str) -> CommandResults:
+def get_file_indicator_reputation(stix_object: dict, reputation_score: int, reliability_score: str,
+                                  tlp: str, is_unknown: bool) -> CommandResults:
     """
     Return stix_object of type file as indicator
     """
 
-    hashes = extract_file_indicator_hashes(stix_object["pattern"])
+    hashes = extract_file_indicator_hashes(stix_object.get("pattern"))
```
Comment (sapirshuker) [inline Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py:288]: You don't return any indicator when you don't have pattern, so you are returning empty hash dict. As a result, we do not have a verdict (I saw it in the last demo). we need to return some value here.

**Review by sapirshuker — CHANGES_REQUESTED** [review]
Great Job!! 
please see my comment
There is not much more to change, we are almost there! 
Thank you very match for you contribution and for your patience.
Please feel free to reach out to me with any questions

File: Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py [hunk 2]
```diff
@@ -166,13 +166,16 @@ def extract_indicator_from_pattern(stix_object: dict) -> str:
     return item[0][2].strip("'")
 
 
-def extract_file_indicator_hashes(pattern_str: str | None) -> dict:
+def extract_file_indicator_hashes(stix_object: str | None) -> dict:
     """
     Extract hashes composing the STIX indicator pattern
     """
     hashes = {"sha512": "", "sha256": "", "sha1": "", "md5": ""}
```
Comment (sapirshuker) [inline Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py:173]:
```suggestion
    hashes = {"sha512": None, "sha256": None, "sha1": None, "md5": None}
```
We don't want to set an empty string to an indicator value.

File: Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py [hunk 3]
```diff
@@ -285,7 +288,7 @@ def get_file_indicator_reputation(stix_object: dict, reputation_score: int, reli
     Return stix_object of type file as indicator
     """
 
-    hashes = extract_file_indicator_hashes(stix_object.get("pattern"))
+    hashes = extract_file_indicator_hashes(stix_object)
     dbot_score = Common.DBotScore(
         indicator=hashes["md5"],
```
Comment (sapirshuker) [inline Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py:293]: What will happen if we don't have the MD5 value (None)? We need to return a value

**Review by sapirshuker — COMMENTED** [review]
Great!! just few minor edge cases!

File: Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py (hunk 3 above)
Comment (PierrickV) [reply Packs/SEKOIAIntelligenceCenter/Integrations/SEKOIAIntelligenceCenter/SEKOIAIntelligenceCenter.py]: Hello @sapirshuker,

What do you think of:
image


Have a nice day

**Review by MLainer1 — COMMENTED** [review]
TIM review complete

**Comment by sapirshuker** [comment]
@PierrickV 
Hey, Great Job! we are almost there!
To continue with the merge to internal PR, I would appreciate it if you could merge from master. 
Thank you!

**Comment by PierrickV** [comment]
Done @sapirshuker

**Review by sapirshuker — APPROVED** [review]
Great Job!!
It's great to finally reach our destination! I enjoyed working with you.

**Pull request merged**